### PR TITLE
[sw/silicon_creator, test] Mask MSb of cpuctrl CSR and ddd manual test

### PR DESCRIPTION
**Pull request by alphan**
Signed-off-by: Alphan Ulusoy <alphan@google.com>

### CODE REVIEW

File: sw/device/silicon_creator/rom/data/rom_manual_testplan.hjson [hunk 1]
```diff
@@ -25,5 +25,27 @@
...
     }
+    {
+      name: rom_manual_cpuctrl_layout
+      desc: '''Verify that the mask for the `cputctrl` CSR in `rom_init()` is up to date.
+
+            See [Ibex documentation][1] for details.
+
+            - Verify that the first six bits of the `CPUCTRL` CSR match what
+              `rom_init()` and `CREATOR_SW_CFG_CPUCTRL` OTP item expect:
+              - Bit 8: `lc_scr_key_valid`
```
Comment (moidx) [inline sw/device/silicon_creator/rom/data/rom_manual_testplan.hjson:36]: nit: `ic_scr_key_valid`

File: sw/device/silicon_creator/rom/rom.c [hunk 2]
```diff
@@ -292,12 +292,13 @@ static void rom_pre_boot_check(void) {
   CFI_FUNC_COUNTER_INCREMENT(rom_counters, kCfiRomPreBootCheck, 4);
 
   // Check the cpuctrl CSR.
-  // Note: We don't mask the CSR value here to include exception flags
-  // (bits 6 and 7) in the check.
   uint32_t cpuctrl_csr;
   uint32_t cpuctrl_otp =
       otp_read32(OTP_CTRL_PARAM_CREATOR_SW_CFG_CPUCTRL_OFFSET);
   CSR_READ(CSR_REG_CPUCTRL, &cpuctrl_csr);
+  // We only mask the 8th bit (`lc_scr_key_valid`) to include exception flags
```
Comment (moidx) [inline sw/device/silicon_creator/rom/rom.c:299]: nit: `ic_scr_key_valid`

File: sw/device/silicon_creator/rom/data/rom_manual_testplan.hjson [hunk 3]
```diff
@@ -25,5 +25,27 @@
...
       tests: []
     }
+    {
+      name: rom_manual_cpuctrl_layout
+      desc: '''Verify that the mask for the `cputctrl` CSR in `rom_init()` is up to date.
+
+            See [Ibex documentation][1] for details.
+
+            - Verify that the first six bits of the `CPUCTRL` CSR match what
+              `rom_init()` and `CREATOR_SW_CFG_CPUCTRL` OTP item expect:
```
Comment (moidx) [inline sw/device/silicon_creator/rom/data/rom_manual_testplan.hjson:35]: nit: TODO cross-reference OTP documentation describing the bit-8 masking behavior.

**Review by moidx — APPROVED** [review]

File: sw/device/silicon_creator/rom/data/rom_manual_testplan.hjson (hunk 3 above)
Comment (alphan) [reply sw/device/silicon_creator/rom/data/rom_manual_testplan.hjson:35]: We actually write only bits 5:0 to `cpuctrl` so I added a comment for this to `otp_ctrl_img_creator_sw_cfg.hjson`. The hardened check that masks bit 8 but includes bit 7 and 6 is an implementation detail that doesn't have to go into the documentation, I think.

**Comment by alphan** [comment]
Merging now since the previous tip passed all checks and the last push was a trivial change.

image

**Pull request merged**